pkg/controller/daemon/update.go

```diff
@@ -194,6 +239,18 @@ func (dsc *DaemonSetsController) rollingUpdate(ctx context.Context, ds *apps.Dae
 	// use any of the candidates we can, including the allowedNewNodes
 	logger.V(5).Info("DaemonSet allowing replacements", "daemonset", klog.KObj(ds), "replacements", len(allowedNewNodes), "maxSurge", maxSurge, "numSurge", numSurge, "candidates", len(candidateNewNodes))
 	remainingSurge := maxSurge - numSurge
+
+	// decide the deletable number of pods that should not be scheduled anymore
+	// if the number of available pods is greater than the number of pods desired, we adopt the difference
+	deletablePodsNumber := isPodAvailable - desiredNumberScheduled
+	if deletablePodsNumber > 0 {
+		if shouldNotRunPodsToDeleteNumber := len(shouldNotRunPodsToDelete); deletablePodsNumber > shouldNotRunPodsToDeleteNumber {
+			deletablePodsNumber = shouldNotRunPodsToDeleteNumber
+		}
+		logger.V(5).Info("DaemonSet pod should be deleted from the node as it is not schedulable anymore", "daemonset", klog.KObj(ds), "deletablePodsNumber", deletablePodsNumber)
```

Comment:
@atiratree 
I'll add the check if `shouldNotRunPodsToDelete` is empty, but is that enough for `deletablePodsNumber`?

https://github.com/mochizuki875/kubernetes/blob/9e41a1fdb1f560726baa05cd425a7e273a79670d/pkg/controller/daemon/update.go#L243-L246

> I think we should also log in a similar format we do above. So best is to iterate over oldPodsToDelete and construct pod obj argument.

Do you mean we should newly emit log for `oldPodsToDelete` like that, after `shouldNotRunPodsToDelete[:deletablePodsNumber]` is appended to `oldPodsToDelete`?

```go
	for _, podToDeleteName := range oldPodsToDelete {
		podToDelete, err := dsc.podLister.Pods(ds.Namespace).Get(podToDeleteName)
		if err != nil {
			return fmt.Errorf("couldn't get pod for podToDelete %q: %v", podToDeleteName, err)
		}
		logger.V(0).Info("DaemonSet pod should be deleted", "daemonset", klog.KObj(ds), "podToDelete", klog.KObj(podToDelete))
	}
```

Reply:
I was just mentioning the cases where it logs, no need to add additional checks.

And I made a mistake we should iterate over shouldNotRunPodsToDelete after we change its size since oldPodsToDelete are already logged before (except the `!shouldRun` we added in this PR which should log as well).